src/arkode/arkode_forcingstep.c

```diff
@@ -234,37 +234,81 @@ static int forcingStep_TakeStep(ARKodeMem ark_mem, sunrealtype* dsmPtr,
 
   SUNStepper s0    = step_mem->stepper[0];
   sunrealtype tout = ark_mem->tn + ark_mem->h;
-  sunrealtype tret = 0;
+  sunrealtype tret = ZERO;
 
   /* Evolve stepper 0 on its own */
+  SUNLogInfo(ARK_LOGGER, "begin-partition", "partition = 0");
+
   SUNErrCode err = SUNStepper_Reset(s0, ark_mem->tn, ark_mem->yn);
-  if (err != SUN_SUCCESS) { return ARK_SUNSTEPPER_ERR; }
+  if (err != SUN_SUCCESS)
+  {
+    SUNLogInfo(ARK_LOGGER, "end-partition",
+               "status = failed stepper reset, err = %i", err);
+    return ARK_SUNSTEPPER_ERR;
+  }
+
   err = SUNStepper_SetStopTime(s0, tout);
-  if (err != SUN_SUCCESS) { return ARK_SUNSTEPPER_ERR; }
+  if (err != SUN_SUCCESS)
+  {
+    SUNLogInfo(ARK_LOGGER, "end-partition",
+               "status = failed set stop time, err = %i", err);
+    return ARK_SUNSTEPPER_ERR;
+  }
+
   err = SUNStepper_Evolve(s0, tout, ark_mem->ycur, &tret);
```

Comment:
Perhaps we can save this for a future PR, but I'd think that it would be good to print `ark_mem->ycur` here when running with "extra debugging" enabled.

Reply:
This is quick and easy to add, I'll put in the logging statements in a sec